src/frontend/src/lib/stores/token.store.ts

```diff
@@ -1,33 +1,53 @@
 import type { OptionToken } from '$lib/types/token';
+import { getContext } from 'svelte';
 import { writable, type Readable } from 'svelte/store';
 
 export type TokenData = OptionToken;
 
 export interface TokenStore extends Readable<TokenData> {
 	set: (token: TokenData) => void;
 	reset: () => void;
+	loadTokenAndRun: ({
+		token,
+		callback
+	}: {
+		token: TokenData;
+		callback: () => Promise<void>;
+	}) => Promise<void>;
+	runAndResetToken: (callback: () => void) => void;
 }
 
-const initTokenStore = (): TokenStore => {
+export const initTokenStore = (): TokenStore => {
 	const INITIAL: TokenData = undefined;
 
 	const { subscribe, set } = writable<TokenData>(INITIAL);
 
+	const loadTokenAndRun = async ({
+		token,
+		callback
+	}: {
+		token: TokenData;
+		callback: () => Promise<void>;
+	}) => {
+		set(token);
+		await callback();
+	};
+
+	const runAndResetToken = (callback: () => void) => {
+		callback();
+		set(null);
+	};
+
 	return {
 		subscribe,
 		set: (token: TokenData) => set(token),
-		reset: () => set(null)
+		reset: () => set(null),
+		loadTokenAndRun,
+		runAndResetToken
 	};
 };
 
-/**
- * Previously, we had two distinct screens: one displaying in which no token was selected (Tokens list) and another showing details for a selected token (Transaction page).
- * However, to provide "Receive" and "Send" actions for each token directly in the token list, we started using the store on this screen as well.
- * This was the most pragmatic approach to implement the feature within a short time frame.
- * Unfortunately, this architecture is not ideal. If the global token store is used for other features on the main screen in the future, conflicts may arise.
- * Therefore, this store is deprecated and should eventually be replaced with a method that passes down the selected token, ideally through contexts.
- * In other words, for the Hero, Receive, and Send features, we should avoid accessing a global store and instead use a dedicated state.
- * TODO: pass down token with context instead of global store.
- * @deprecated This approach works for now but does not align with the new architectural requirements.
- */
-export const token = initTokenStore();
+export const TOKEN_CONTEXT_KEY = Symbol('token');
+
+export const getTokenStoreFromContext = (): TokenStore =>
```

Comment:
I would not proceed this with a big bang migration but, I've got the feeling that it would be easier to introduce the context piece by piece. For example there are already 40 changes in this PR, that's quite a lot. Introducing a new context might help providing smaller PR.